<a id="TableOfContents"></a>
# TABLE OF CONTENTS:
<li><a href='#imports'>Imports</a></li>
<li><a href="#Q1">Question 1</a></li>
<li><a href='#Q2'>Question 2</a></li>
<li><a href='#Q3'>Question 3</a></li>

Let's set up an example scenario as perspective for our regression exercises using the Zillow dataset.

As a Codeup data science graduate, you want to show off your skills to the Zillow data science team in hopes of getting an interview for a position you saw pop up on LinkedIn. You thought it might look impressive to build an end-to-end project in which you use some of their Kaggle data to predict property values using some of their available features; who knows, you might even do some feature engineering to blow them away. Your goal is to predict the values of single unit properties using the obervations from 2017.

In these exercises, you will complete the first step toward the above goal: acquire and prepare the necessary Zillow data from the zillow database in the Codeup database server.

<a id='imports'></a>
# IMPORTS:
<li><a href='#TableOfContents'>Table of Contents</a></li>

In [1]:
# Vectorization and tables
import numpy as np
import pandas as pd

# Visualization
import matplotlib.pyplot as plt
import seaborn as sns

# Stats
from scipy import stats

# Connect to sql server
import env
import wrangle

<a id='Q1'></a>
# Question 1
<li><a href='#TableOfContents'>Table of Contents</a></li>

### 1. Acquire bedroomcnt, bathroomcnt, calculatedfinishedsquarefeet, taxvaluedollarcnt, yearbuilt, taxamount, and fips from the zillow database for all 'Single Family Residential' properties.

In [2]:
zillow = wrangle.acquire()

In [3]:
zillow.sample(5)

bedroomcnt  bathroomcnt  calculatedfinishedsquarefeet  \
1442147         3.0          2.5                        2036.0   
146790          3.0          2.0                        1718.0   
410467          3.0          2.0                        1032.0   
714061          5.0          7.0                        9828.0   
395726          3.0          1.0                        1030.0   

         taxvaluedollarcnt  yearbuilt  taxamount    fips  
1442147           514072.0     1978.0    5149.92  6059.0  
146790            490000.0     1964.0    6160.78  6037.0  
410467             69534.0     1938.0     951.48  6037.0  
714061           7637390.0     1934.0   91860.96  6037.0  
395726            310660.0     1925.0    3989.00  6059.0

<a id='Q2'></a>
# Question 2
<li><a href='#TableOfContents'>Table of Contents</a></li>

### 2. Using your acquired Zillow data, walk through the summarization and cleaning steps in your wrangle.ipynb file like we did above. You may handle the missing values however you feel is appropriate and meaningful; remember to document your process and decisions using markdown and code commenting where helpful.

### Issues To Fix:

- Column Types
    - bedroomcnt
        - Float to int(19 - categorical)
        - 1-16, 18, 25
    - bathroomcnt
        - Float(38 - categorical)?
        - 0-20, 32
        - 0.5-12.5, 14.5, 19.5
        - 1.75
    - taxvaluedollarcount
        - Float(10580 - categorical)?
    - yearbuilt
        - Float to int(153 - categorical)?
    - fips
        - Float to object(3 - categorical)
        - 6037, 6059, 6111
- Column Values
    - bedroomcnt
        - 11 na
        - Fill with mode
    - bathroomcnt
        - 11 na
        - Fill with mode
    - calculatedfinishedsquarefeet
        - 8484 na
        - Fill with mean
    - taxvaluedollarcount
        - 493 na
        - Fill with mean
    - yearbuilt
        - 9337 na
        - Fill with ?
    - taxamount
        - 4442 na
        - Fill with mean

In [5]:
zillow.dtypes

bedroomcnt                      float64
bathroomcnt                     float64
calculatedfinishedsquarefeet    float64
taxvaluedollarcnt               float64
yearbuilt                       float64
taxamount                       float64
fips                            float64
dtype: object

In [34]:
zillow.fips.value_counts()

6037.0    1431811
6059.0     555077
6111.0     165975
Name: fips, dtype: int64

<a id='Q3'></a>
# Question 3
<li><a href='#TableOfContents'>Table of Contents</a></li>

### 3. Store all of the necessary functions to automate your process from acquiring the data to returning a cleaned dataframe with no missing values in your wrangle.py file. Name your final function wrangle_zillow.